### PR TITLE
improve(eslint): remove gulp-eslint

### PR DESCRIPTION
**Pull request by tqmukas**

### CODE REVIEW

File: lib/tasks/eslint.js [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-  return new Promise((resolve, reject) =>
-    gulp.src(files)
-      .pipe(gulpEslint())
-      .pipe(gulpEslint.format())
-      .pipe(gulpEslint.failAfterError())
-      .on('error', reject)
-      .once('end', resolve)
-  );
+function eslint({dirs}) {
+  return Promise.resolve().then(() => {
```
Comment (ronami) [inline lib/tasks/eslint.js]: Wouldn't it be clearer to use new Promise() instead?

File: lib/tasks/eslint.js [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
+    const files = dirs || globs.eslint();
```
Comment (ronami) [inline lib/tasks/eslint.js]: Use default function params

File: lib/tasks/eslint.js [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
+    const cli = new CLIEngine();
```
Comment (ronami) [inline lib/tasks/eslint.js]: Can we create CLIEngine once and re-use it?

File: lib/tasks/eslint.js (hunk 1 above)
Comment (tqmukas) [inline lib/tasks/eslint.js]: There is this pretty known tactic in Promise world to wrap you code inside `Promise.resolve().then()` so it works as sort of `try/catch`. This means all the errors inside the callback will go to `catch`.

File: lib/tasks/eslint.js (hunk 2 above)
Comment (tqmukas) [inline lib/tasks/eslint.js]: Can you explain what do you mean?

File: lib/tasks/eslint.js (hunk 3 above)
Comment (tqmukas) [inline lib/tasks/eslint.js]: Yeah just we would loose the `try/catch` safety check. The on that I mentioned above.

File: lib/tasks/eslint.js (hunk 1 above)
Comment (tqmukas) [inline lib/tasks/eslint.js]: By the way, eslint has only synchronous api. No need to use `new Promise`.

**Pull request merged**